Add k8s 1.25 variants

stmcginnis opened the pull request:
**Issue number:**

Closes #2698

**Description of changes:**

This adds a k8s-1.25 package.
Existing 1.24 models are moved to be 1.25, with symlinks created for 1.24 models to point to the new locations.
Finally, new variants are defined for aws-k8s-1.25, aws-k8s-1.25-nvidia, metal-k8s-1.25, and vmware-k8s-1.25.

**Testing done:**

To be done:

- [x] Update package with final EKS-D 1.25 package source
- [x] Test aws-k8s-1.25 x86_64 joining 1.25 cluster and ran full conformance test
```
12:04:16       e2e   global   complete   passed   Passed:362, Failed:  0                                                                 
12:04:16 Sonobuoy has completed. Use `sonobuoy retrieve` to get results. 
```
- [x] Test aws-k8s-1.25 ARM joining 1.25 cluster and deploying pods
- [x] Test aws-k8s-1.25-nvidia x86-64 joining 1.25 cluster and ran full conformance test
```
      "progress": {
        "name": "e2e",
        "node": "global",
        "timestamp": "2023-01-13T21:45:38.815114657Z",
        "msg": "Test Suite completed",
        "total": 0,
        "completed": 362
      }
```
- [x] Test aws-k8s-1.25-nvidia ARM joining 1.25 cluster and ran full conformance test
- [x] Test metal-k8s-1.25 joining 1.24 cluster and ran full conformance test
```
Plugin: e2e
Status: passed
Total: 6973
Passed: 356
Failed: 0
Skipped: 6617
```
- [x] Test vmware-k8s-1.25 joining 1.24 cluster and ran full conformance test
```
      "progress": {
        "name": "e2e",
        "node": "global",
        "timestamp": "2023-01-13T23:27:27.38052641Z",
        "msg": "Test Suite completed",
        "total": 356,
        "completed": 356
      }
```
- [x] Check kubelet logs and system logs in general, compared them with the aws-k8s-1.24 variant kubelet logs.

<details>

Note that there are a few flags that are being deprecated. We are planning migrating to `--cloud-provider external` for 1.26 and beyond.

```bash
bash-5.1# journalctl --no-hostname -u  kubelet
Jan 13 18:23:49 systemd[1]: Starting Kubelet...
Jan 13 18:23:50 host-ctr[1150]: time="2023-01-13T18:23:50Z" level=info msg="Image does not exist, proceeding to pull image from source." ref="ecr.aws/arn:aws:ecr:us-west-2:602401143452:repository/eks/pause:3.1-eksbuild.1"
Jan 13 18:23:50 host-ctr[1150]: time="2023-01-13T18:23:50Z" level=info msg="pulling with Amazon ECR Resolver" ref="ecr.aws/arn:aws:ecr:us-west-2:602401143452:repository/eks/pause:3.1-eksbuild.1"
Jan 13 18:23:50 host-ctr[1150]: time="2023-01-13T18:23:50Z" level=info msg="pulled image successfully" img="ecr.aws/arn:aws:ecr:us-west-2:602401143452:repository/eks/pause:3.1-eksbuild.1"
Jan 13 18:23:50 host-ctr[1150]: time="2023-01-13T18:23:50Z" level=info msg="unpacking image..." img="ecr.aws/arn:aws:ecr:us-west-2:602401143452:repository/eks/pause:3.1-eksbuild.1"
Jan 13 18:23:51 host-ctr[1150]: time="2023-01-13T18:23:51Z" level=info msg="tagging image" img="602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/pause:3.1-eksbuild.1"
Jan 13 18:23:53 kubelet[1165]: Flag --cloud-provider has been deprecated, will be removed in 1.25 or later, in favor of removing cloud provider code from Kubelet.
Jan 13 18:23:53 kubelet[1165]: Flag --container-runtime has been deprecated, will be removed in 1.27 as the only valid value is 'remote'
Jan 13 18:23:53 kubelet[1165]: Flag --containerd has been deprecated, This is a cadvisor flag that was mistakenly registered with the Kubelet. Due to legacy concerns, it will follow the standard CLI deprecation timeline before being removed.
Jan 13 18:23:53 kubelet[1165]: Flag --register-with-taints has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Jan 13 18:23:53 kubelet[1165]: Flag --pod-infra-container-image has been deprecated, will be removed in 1.27. Image garbage collector will get sandbox image information from CRI.
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.388484    1165 server.go:200] "--pod-infra-container-image will not be pruned by the image garbage collector in kubelet and should also be set in the remote runtime"
Jan 13 18:23:53 kubelet[1165]: Flag --cloud-provider has been deprecated, will be removed in 1.25 or later, in favor of removing cloud provider code from Kubelet.
Jan 13 18:23:53 kubelet[1165]: Flag --container-runtime has been deprecated, will be removed in 1.27 as the only valid value is 'remote'
Jan 13 18:23:53 kubelet[1165]: Flag --register-with-taints has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Jan 13 18:23:53 kubelet[1165]: Flag --pod-infra-container-image has been deprecated, will be removed in 1.27. Image garbage collector will get sandbox image information from CRI.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.393352    1165 feature_gate.go:237] Setting GA feature gate CSIMigration=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.393365    1165 feature_gate.go:237] Setting GA feature gate CSIMigrationAWS=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.393653    1165 feature_gate.go:237] Setting GA feature gate CSIMigration=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.393681    1165 feature_gate.go:237] Setting GA feature gate CSIMigrationAWS=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.397475    1165 server.go:413] "Kubelet version" kubeletVersion="v1.25.5-eks-c248520"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.397497    1165 server.go:415] "Golang settings" GOGC="" GOMAXPROCS="" GOTRACEBACK=""
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.397563    1165 feature_gate.go:237] Setting GA feature gate CSIMigration=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.397574    1165 feature_gate.go:237] Setting GA feature gate CSIMigrationAWS=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.397669    1165 feature_gate.go:237] Setting GA feature gate CSIMigration=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.397677    1165 feature_gate.go:237] Setting GA feature gate CSIMigrationAWS=true. It will be removed in a future release.
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.397814    1165 plugins.go:132] WARNING: aws built-in cloud provider is now deprecated. The AWS provider is deprecated and will be removed in a future release. Please use https://github.com/kubernetes/cloud-provider-aws
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.400465    1165 aws.go:1228] Get AWS region from metadata client
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.400758    1165 aws.go:1271] Zone not specified in configuration file; querying AWS metadata service
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.411070    1165 aws.go:1311] Building AWS cloudprovider
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.638511    1165 tags.go:80] AWS cloud filtering on ClusterID: br-1-25-ipv4-beta
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.659844    1165 dynamic_cafile_content.go:157] "Starting controller" name="client-ca-bundle::/etc/kubernetes/pki/ca.crt"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.694258    1165 container_manager_linux.go:262] "Container manager verified user specified cgroup-root exists" cgroupRoot=[]
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.694416    1165 container_manager_linux.go:267] "Creating Container Manager object based on Node Config" nodeConfig={RuntimeCgroupsName: SystemCgroupsName: KubeletCgroupsName: KubeletOOMScoreAdj:-999 ContainerRuntime: CgroupsPerQOS:true CgroupRoot:/ CgroupDriver:systemd KubeletRootDir:/var/lib/kubelet ProtectKernelDefaults:true NodeAllocatableConfig:{KubeReservedCgroupName:/runtime SystemReservedCgroupName: ReservedSystemCPUs: EnforceNodeAllocatable:map[pods:{}] KubeReserved:map[cpu:{i:{value:70 scale:-3} d:{Dec:<nil>} s:70m Format:DecimalSI} ephemeral-storage:{i:{value:1073741824 scale:0} d:{Dec:<nil>} s:1Gi Format:BinarySI} memory:{i:{value:601882624 scale:0} d:{Dec:<nil>} s:574Mi Format:BinarySI}] SystemReserved:map[] HardEvictionThresholds:[{Signal:imagefs.available Operator:LessThan Value:{Quantity:<nil> Percentage:0.15} GracePeriod:0s MinReclaim:<nil>} {Signal:memory.available Operator:LessThan Value:{Quantity:100Mi Percentage:0} GracePeriod:0s MinReclaim:<nil>} {Signal:nodefs.available Operator:LessThan Value:{Quantity:<nil> Percentage:0.1} GracePeriod:0s MinReclaim:<nil>} {Signal:nodefs.inodesFree Operator:LessThan Value:{Quantity:<nil> Percentage:0.05} GracePeriod:0s MinReclaim:<nil>}]} QOSReserved:map[] ExperimentalCPUManagerPolicy:none ExperimentalCPUManagerPolicyOptions:map[] ExperimentalTopologyManagerScope:container ExperimentalCPUManagerReconcilePeriod:10s ExperimentalMemoryManagerPolicy:None ExperimentalMemoryManagerReservedMemory:[] ExperimentalPodPidsLimit:-1 EnforceCPULimits:true CPUCFSQuotaPeriod:100ms ExperimentalTopologyManagerPolicy:none}
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.694443    1165 topology_manager.go:134] "Creating topology manager with policy per scope" topologyPolicyName="none" topologyScopeName="container"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.694460    1165 container_manager_linux.go:302] "Creating device plugin manager" devicePluginEnabled=true
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.695194    1165 state_mem.go:36] "Initialized new in-memory state store"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.724175    1165 kubelet.go:381] "Attempting to sync node with API server"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.724303    1165 kubelet.go:270] "Adding static pod path" path="/etc/kubernetes/static-pods/"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.724402    1165 kubelet.go:281] "Adding apiserver pod source"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.724500    1165 apiserver.go:42] "Waiting for node sync before watching apiserver pods"
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.726568    1165 kuberuntime_manager.go:240] "Container runtime initialized" containerRuntime="containerd" version="1.6.15+bottlerocket" apiVersion="v1"
Jan 13 18:23:53 kubelet[1165]: W0113 18:23:53.731691    1165 probe.go:268] Flexvolume plugin directory at /var/lib/kubelet/plugins/volume/exec does not exist. Recreating.
Jan 13 18:23:53 kubelet[1165]: I0113 18:23:53.732684    1165 server.go:1175] "Started kubelet"
Jan 13 18:23:53 systemd[1]: Started Kubelet.
...
```

</details>

- [x] Confirm the vSphere CSI driver still works
- [x] Confirm the EBS CSI driver still works.
- [x] Smoke test aws-k8s-1.25-nvidia variant with some GPU workloads

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

stmcginnis committed:
Add k8s 1.25 variants

This adds a k8s-1.25 package. Existing 1.24 models are moved to be 1.25,
with symlinks created for 1.24 models to point to the new locations.
Finally, new variants are defined for aws-k8s-1.25, aws-k8s-1.25-nvidia,
metal-k8s-1.25, and vmware-k8s-1.25.

Signed-off-by: Sean McGinnis <stmcg@amazon.com>

etungsten commented:
We need to remember to merge in changes from https://github.com/bottlerocket-os/bottlerocket/pull/2697 once that's done and apply the same OCI defaults for 1.25.

etungsten commented on packages/kubernetes-1.25/kubelet-exec-start-conf:
```diff
@@ -0,0 +1,31 @@
+    --bootstrap-kubeconfig /etc/kubernetes/kubelet/bootstrap-kubeconfig \
+{{/if}}
+{{else}}
+    --cloud-provider "" \
```
I think we need to migrate this to `--cloud-provider=external` to move away from in-tree AWS cloud provider and start depending on EKS' cloud provider add-on/plugin

etungsten replied:
Scratch all that, we need to keep this empty to not enable a cloud-provider for metal variants where a cloud manager might not exist.

etungsten reviewed (commented):
Can we please add the following to the testing checklist?

- [ ] Check kubelet logs and system logs in general, compared them with the aws-k8s-1.24 variant kubelet logs.
- [ ] Confirm the vSphere CSI driver still works
- [ ] Confirm the EBS CSI driver still works.
- [ ] Smoke test aws-k8s-1.25-nvidia variant with some GPU workloads

Basically the list of testing done in https://github.com/bottlerocket-os/bottlerocket/pull/2140

etungsten commented on packages/kubernetes-1.25/kubelet-exec-start-conf:
```diff
@@ -0,0 +1,31 @@
+ExecStart=
+ExecStart=/usr/bin/kubelet \
+{{#unless settings.kubernetes.standalone-mode}}
+    --cloud-provider {{default "external" settings.kubernetes.cloud-provider}} \
```
Sorry, this is actually the place we need to change. We can just use `--cloud-provider=external` for all variants since in-tree cloud provider code is being completely removed from `kubelet` and for aws we're temporary migrating to `external` so it can depend on the external EKS-provided cloud-node controller.

This would also mean `settings.kubernetes.cloud-provider` is going to be an inert setting that doesn't do anything, but that's probably fine? @bcressey

etungsten replied:
Or we can choose to change the default setting here:
https://github.com/bottlerocket-os/bottlerocket/blob/b5b6540cc96891548744c9e2041b12c03922fa86/sources/models/shared-defaults/kubernetes-aws.toml#L6

to "external" so we get to keep the setting until the time when kubelet actually removes the `--cloud-provider` flag.

stmcginnis replied:
I think we need to change it to be explicit in the `kubelet-exec-start-conf` file. All k8s variants use the `shared-defaults` settings, so if we changed that to `external`, the earlier versions of k8s that should still be using in-tree providers would pick that up.

etungsten replied:
> All k8s variants use the shared-defaults settings

Ah, I almost forgot about that. You're absolutely right.

stmcginnis commented:
Testing results for `aws-k8s-1.25-x86_64`:

Using IPv4:

```txt
> sonobuoy status --kubeconfig br-1-25-ipv4-x86-64-beta.kubeconfig
   PLUGIN     STATUS   RESULT   COUNT                 PROGRESS
      e2e   complete   passed       1                 Passed:362, Failed:  0
```

Using IPv6:

```txt
> sonobuoy status --kubeconfig br-1-25-ipv6-x86-64-beta.kubeconfig
   PLUGIN     STATUS   RESULT   COUNT                 PROGRESS
      e2e   complete   passed       1   Passed:362, Failed:  0
```

NVIDIA variant:

```txt
> sonobuoy status --kubeconfig br-1-25-nvidia-x86-64-beta.kubeconfig
   PLUGIN     STATUS   RESULT   COUNT                 PROGRESS
      e2e   complete   passed       1   Passed:362, Failed:  0
```

and NVIDIA smoke tests:

```txt
...
90      images/sec: 75.5 +/- 0.1 (jitter = 0.5) 7.596
100     images/sec: 75.5 +/- 0.0 (jitter = 0.5) 8.053
----------------------------------------------------------------
total images/sec: 75.37
----------------------------------------------------------------
```

stmcginnis commented:
Latest updates:

- Rebased on `develop` to resolve conflicts with autoscaling updates
- Reverted kubelet startup config to use configured in-tree cloud provider for now
- Updated k8s-1.25 EKS-D package to latest rev

etungsten commented on sources/models/src/vmware-k8s-1.25/defaults.d/26-aws-autoscaling.toml:
```diff
@@ -0,0 +1 @@
+../../../shared-defaults/aws-autoscaling.toml
```
We need to remove this from vmware-k8s. The autoscaling settings do not exist in vmware-k8s-1.25 variants.

stmcginnis replied:
Done!

etungsten commented:
We can reconstruct the symlinks again since `*k8s-1.23*`, `*k8s-1.24*`, `*k8s-1.25*` share defaults and settings model:

```bash
❯ ls -al sources/models/src
total 64
drwxr-xr-x. 1 fedora fedora   806 Jan 13 23:43 .
drwxr-xr-x. 1 fedora fedora   202 Jan 13 23:43 ..
drwxr-xr-x. 1 fedora fedora    32 Nov 14 19:25 aws-dev
drwxr-xr-x. 1 fedora fedora    32 Nov  9 01:28 aws-ecs-1
drwxr-xr-x. 1 fedora fedora    32 Nov  9 01:28 aws-ecs-1-nvidia
lrwxrwxrwx. 1 fedora fedora    14 Nov  9 01:28 aws-k8s-1.21 -> ./aws-k8s-1.22
lrwxrwxrwx. 1 fedora fedora    21 Nov  9 01:28 aws-k8s-1.21-nvidia -> ./aws-k8s-1.22-nvidia
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 aws-k8s-1.22
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 aws-k8s-1.22-nvidia
lrwxrwxrwx. 1 fedora fedora    12 Nov  9 01:28 aws-k8s-1.23 -> aws-k8s-1.24
lrwxrwxrwx. 1 fedora fedora    19 Nov  9 01:28 aws-k8s-1.23-nvidia -> aws-k8s-1.24-nvidia
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 aws-k8s-1.24
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 aws-k8s-1.24-nvidia
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 aws-k8s-1.25
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 aws-k8s-1.25-nvidia
-rw-r--r--. 1 fedora fedora  7291 Nov  9 01:28 de.rs
-rw-r--r--. 1 fedora fedora  2741 Nov  9 01:28 exec.rs
-rw-r--r--. 1 fedora fedora 15551 Jan 13 23:43 lib.rs
drwxr-xr-x. 1 fedora fedora    32 Nov  9 01:28 metal-dev
lrwxrwxrwx. 1 fedora fedora    16 Nov  9 01:28 metal-k8s-1.21 -> ./metal-k8s-1.22
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 metal-k8s-1.22
lrwxrwxrwx. 1 fedora fedora    14 Nov  9 01:28 metal-k8s-1.23 -> metal-k8s-1.24
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 metal-k8s-1.24
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 metal-k8s-1.25
drwxr-xr-x. 1 fedora fedora    68 Jan 13 23:43 modeled_types
drwxr-xr-x. 1 fedora fedora    10 Nov  9 01:28 variant
-rw-r--r--. 1 fedora fedora   815 Nov  9 01:28 variant_mod.rs
drwxr-xr-x. 1 fedora fedora    32 Nov 14 19:25 vmware-dev
lrwxrwxrwx. 1 fedora fedora    17 Nov  9 01:28 vmware-k8s-1.21 -> ./vmware-k8s-1.22
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 vmware-k8s-1.22
lrwxrwxrwx. 1 fedora fedora    15 Nov  9 01:28 vmware-k8s-1.23 -> vmware-k8s-1.24
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 vmware-k8s-1.24
drwxr-xr-x. 1 fedora fedora    32 Jan 13 23:43 vmware-k8s-1.25
```

etungsten commented:
We are now missing `../../../shared-defaults/aws-autoscaling.toml` from `shared-defaults` for `aws-k8s-1.25` and `aws-k8s-1.25-nvidia`.

stmcginnis commented:
> We are now missing `../../../shared-defaults/aws-autoscaling.toml` from `shared-defaults` for `aws-k8s-1.25` and `aws-k8s-1.25-nvidia`.

I now realize I completely mis-read your previous comment and instead of removing from `vmware-k8s-*` I removed it from all 1.25 variants. Will get that updated!

bcressey commented on sources/models/src/vmware-k8s-1.25/vmware-k8s-1.24:
```diff
@@ -0,0 +1 @@
+vmware-k8s-1.24
```
this is a dead symlink, but isn't needed

stmcginnis replied:
That's where that went!

bcressey commented on variants/aws-k8s-1.25/Cargo.toml:
```diff
@@ -0,0 +1,41 @@
+cni = { path = "../../packages/cni" }
+cni-plugins = { path = "../../packages/cni-plugins" }
+kernel-5_15 = { path = "../../packages/kernel-5.15" }
+kubernetes-1_25= { path = "../../packages/kubernetes-1.25" }
```
nit:
```suggestion
kubernetes-1_25 = { path = "../../packages/kubernetes-1.25" }
```

bcressey commented on packages/kubernetes-1.25/kubernetes-1.25.spec:
```diff
@@ -0,0 +1,162 @@
+# base Apache-2.0, third_party Apache-2.0 AND BSD-3-Clause
+License: Apache-2.0 AND BSD-3-Clause
+URL: https://%{goimport}
+# TODO: Update the following once it is finalized
+Source0: https://distro.eks.amazonaws.com/kubernetes-1-25/releases/2/artifacts/kubernetes/v%{gover}/kubernetes-src.tar.gz
```
LGTM once this is finalized

etungsten commented:
Did some quick tests with an AMI with the latest changes:

Node comes up fine:
```
$ kubectl --kubeconfig etung-br-1-25-ipv4-x86-64.kubeconfig get nodes -o wide
NAME                                            STATUS   ROLES    AGE     VERSION               INTERNAL-IP       EXTERNAL-IP      OS-IMAGE                                KERNEL-VERSION   CONTAINER-RUNTIME
ip-192-168-162-160.us-west-2.compute.internal   Ready    <none>   6m46s   v1.25.5-eks-c248520   192.168.162.160   54.184.206.116   Bottlerocket OS 1.12.0 (aws-k8s-1.25)   5.15.79          containerd://1.6.15+bottlerocket
ip-192-168-74-251.us-west-2.compute.internal    Ready    <none>   6m47s   v1.25.5-eks-c248520   192.168.74.251    35.161.142.225   Bottlerocket OS 1.12.0 (aws-k8s-1.25)   5.15.79          containerd://1.6.15+bottlerocket
```
Pods run fine:
```
$ kubectl --kubeconfig etung-br-1-25-ipv4-x86-64.kubeconfig get pods -A
NAMESPACE     NAME                       READY   STATUS    RESTARTS   AGE
kube-system   aws-node-6r2pk             1/1     Running   0          6m57s
kube-system   aws-node-mwmxq             1/1     Running   0          6m56s
kube-system   coredns-554d8bdcfd-flzqs   1/1     Running   0          3h17m
kube-system   coredns-554d8bdcfd-sdwmr   1/1     Running   0          3h17m
kube-system   kube-proxy-md7dn           1/1     Running   0          6m56s
kube-system   kube-proxy-t9zrh           1/1     Running   0          6m57s

```
